app/main.py

```diff
@@ -1,8 +1,34 @@
+
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+    @classmethod
+    def sett_partner(cls, human: dict) -> None:
+        person_data = [(key, value) for key, value in human.items()]
+        person_name = person_data[0][1]
+        partner_role = person_data[2][0]
+        partner_name = person_data[2][1]
```

Comment:
It is a hard-coded part in this method that is not descriptive and makes the code less maintainable.